ext/pdo_dblib/dblib_stmt.c

```diff
@@ -125,7 +125,16 @@ static int pdo_dblib_stmt_next_rowset_no_cancel(pdo_stmt_t *stmt)
 	pdo_dblib_db_handle *H = S->H;
 	RETCODE ret;
 
-	ret = dbresults(H->link);
+	int num_fields;
+	do {
+		ret = dbresults(H->link);
+		num_fields = dbnumcols(H->link);
+	} while (H->skip_empty_rowsets && num_fields <= 0 && ret == SUCCEED);
+
+
+	if (H->skip_empty_rowsets && num_fields <= 0) {
```

Comment:
I think it would be better to move this check after the one for FAIL == ret. There might be a situation where you could wind up suppressing a desired error message. The return values would otherwise be the same.

Reply:
I guess it could happen (in case of en empty rowset with an error), I just moved the check after the two others.
From my understanding this new check is a failsafe, in case of an empty rowset and a "ret" value different from FAIL and NO_MORE_RESULTS, so it might as well be the last one...